.github/workflows/dev.yml

```diff
@@ -248,3 +273,39 @@ jobs:
 
       - name: Test
         run: cd build && ctest -j3 --output-on-failure
+
+  heron:
+    # Job to verify that the tasks performed by PrepareRelease have been done. It is
+    # the committer's responsibility (currently) to run PrepareRelease themselves when
+    # making a PR, so that everything is kept in-sync.
+    name: Check autogenerated file freshness
+    runs-on: ubuntu-24.04 # TODO: Update to ubuntu-latest when that switches to 24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: false
+          fetch-depth: 0
+          fetch-tags: false
+
+      - name: PrepareRelease
+        run: maint/PrepareRelease
+
+      - name: 'Rebuild *.h.generic'
+        run: |
+          ./autogen.sh && ./configure
+          rm -f src/*.generic
+          make src/config.h.generic src/pcre2.h.generic
+
+      - name: Working directory clean
+        run:  |
+          if [ -n "`git status --porcelain`" ] ; then
+            (
+              echo "Dirty working tree! Affected files:"
+              git status --porcelain || true
+              echo ""
+              echo "Diff:"
+              git diff || true
```

Comment:
You'll now get build failures if you update documentation files, but forget to sync the HTML or TXT.

Reply:
unless we improve the tooling, this sounds a little restrictive.

Truth is that with Philip being the only native English speaker, his "final" touches over the generated documentation were an invaluable part of making sure it is as high quality as it is (including of course, updating the date with the necessary "fixes")

Reply:
Updating the docs can still be done by me (or other native English speaker). But there seems to be no good use-case for having the different copies of the documentation be out-of-sync with each other. If that ever happens, it's simply because a contributor forgot to sync the HTML or TXT files after updating the man pages.

If you'd prefer, we make it so that the CI job applies the changes and pushes a bot-authored commit to the PR. That seemed like a bit too much infrastructure though, for a simple sanity-check.

Reply:
I'm aware of it. It hosts `index.md` (which is a duplicate of README.md), and the HTML files in `docs/`.

We _could_ do what some projects do, and have different versions of the docs available (for each library version) so the user has a choice of whether to read the "master" or "10.44" docs.

In a future PR, when I spruce up that website a bit, I'll probably just make it so it shows the docs for the last-released version of PCRE2.